keyform/templates/keyform/home.html

```diff
@@ -62,4 +62,40 @@ <h2 class="title">{% trans 'Keyform Requests' %}</h2>
             </a>
         {% endfor %}
     </div>
+
+    {% if is_paginated %}
+        <div class='page-bar'>
+            <ul>
+                <li class='page'><a href="?page=1">First</a></li>
+                {% if page_obj.has_previous %}
+                    <li class='page'><a href="?page={{ page_obj.previous_page_number }}">&laquo;</a></li>
+                {% else %}
+                    <li class="disabled page"><span>&laquo;</span></li>
+                {% endif %}
+                {% if page_obj.number >= 3 %}
+                    <li class='page'><a href="?page={{ page_obj.number|add:"-2" }}">{{ page_obj.number|add:"-2" }}</a></li>
+                {% endif %}
+                {% if page_obj.has_previous %}
+                    <li class="page"><a href="?page={{ page_obj.previous_page_number }}">{{ page_obj.previous_page_number }}</a></li>
+                {% endif %}
+                {% for i in paginator.page_range %}
+                    {% if page_obj.number == i %}
+                        <li class="active page"><span class='current'>{{ i }}</span></li>
+                    {% endif %}
+                {% endfor %}
+                {% if page_obj.has_next %}
+                    <li class="page"><a href="?page={{ page_obj.next_page_number }}">{{ page_obj.next_page_number }}</a></li>
+                {% else %}
+                <li class="disabled page"><span> &raquo;</span></li>
+                {% endif %}
+                {% if page_obj.number <= paginator.num_pages|add:"-2" %}
+                    <li class='page'><a href="?page={{ page_obj.number|add:"2" }}">{{ page_obj.number|add:2 }}</a></li>
+                {% endif %}
+                {% if page_obj.has_next %}
+                    <li class='page'><a href="?page={{ page_obj.next_page_number }}"><span>&raquo;</span></a></li>
+                {% endif %}
+                <li class='page'><a href="?page={{ paginator.num_pages }}">Last</a></li>
```

Comment:
Change this link to point to `?page=last` instead of `?page={{ paginator.num_pages }}` for nicer hotlinking purposes.